app/src/modals/EmptyCompetitionModal/EmptyCompetitionModal.jsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from '../../components/Button';
+import './EmptyCompetitionModal.scss';
+
+function EmptyCompetitionModal({ onConfirm, hideModal }) {
```

Comment:
This hideModal action should be on the page that mounts this modal, like the others modals (check the ImportalPlayersModal) for reference.

The page should have the responsibility to control (hide/show) it's children. Makes the flow easier, this way components/modal can just exist, without having to worry about showing or not.

Reply:
Not quite sure if I understood correctly, but I tried to mimic the way ImportPlayersModal is done.

Reply:
ImportPlayersModal doesnt have a hide callback:

`function ImportPlayersModal({ onConfirm, onClose })`

The hiding action part is handled by the CompetitionPage, whenever onConfirm or onCancel get called

Reply:
I think they both did hiding modal the same way, I had just named them different? I changed the naming scheme.
Or are you talking about handleConfirm()?